File: 005 - LongestPalindromicSubstr/Solution.py
class Solution(object):
    def longestPalindrome(self, s):
        """
        :type s: str
        :rtype: str
        """
        if len(s) <=1:
            return s
        longest = ""
        for i in range(0, len(s)):
            odd = self.return_palindrome(s, i, i)
            even = self.return_palindrome(s, i, i+1)
            longest = max(longest, odd, even, key=len)
        return longest

    def return_palindrome(self, s, left, right):
        """
        :type s: str
        :type left: int
        :type right: int
        :rtype: str
        """
        while(left >=0 and right < len(s) and s[left]==s[right]):
            left-=1
            right+=1
        return s[left+1:right]

File: 005 - LongestPalindromicSubstr/Solution.cpp
#include <iostream>
#include <string>
using namespace std;

class Solution {
public:
    string longestPalindrome(const string& s) {
        string longest = "";
        for (int i=0; i < s.length(); i++){
            string odd = returnPalindrome(s, i ,i);
            string even = returnPalindrome(s, i, i+1);
            string substring = (odd.length()>even.length()) ? odd : even;
            longest = (substring.length()>longest.length()) ? substring : longest;
        }
        return longest;
    }

    string returnPalindrome(const string& s, int left, int right){
        while (left >= 0 && right < s.length() && s[left]==s[right]){
            left--;
            right++;
        }
        return(s.substr(left+1, right-left-1));
    }
};
